README.md
# pjax-table

[![Build Status](https://travis-ci.org/50onRed/pjax-table.svg?branch=master)](https://travis-ci.org/50onRed/pjax-table)

pjax-table is a jQuery plugin that uses [jquery-pjax](https://github.com/defunkt/jquery-pjax) to load server rendered tables and provides table controls for sorting, pagination, row selection, and more.

## Features
  - pjax loading with push state
  - sorting
  - pagination
  - search filtering
  - row selection and manipulation
  - plugin support


## Documentation
The documentation for pjax-table is available [through github pages](http://50onred.github.io/pjax-table/) and is available in the docs folder.


### Building
    // install node dependencies, most notably gulp and gulp plugins
    npm install

    // install client dependencies
    bower install
    
    // after making changes, bump and appropriate version
    // see [gulp-bump](https://github.com/stevelacy/gulp-bump) for semver options
    gulp bump-patch 
    
    // (default task) cleans and then builds standalone and distributable versions with the new version
    gulp



CHANGELOG.md
# Change Log
All notable changes to this project will be documented in this file.
This project adheres to [Semantic Versioning](http://semver.org/).

## [Unreleased][unreleased]

# [2.1.4] - 2018-01-10
### Changed
- Fixed editable action suppression for invalid fields
- editable_cell.js (and editable_cell.min.js) are now separate build artifacts

# [2.1.3] - 2018-01-05
### Changed
- Fix for pagination dropdown
- Update FA icon class for fixed-width

# [2.1.2] - 2017-11-09
### Changed
- Searchbox fix for Bootstrap v4 `hidden` changes
- Bump copyright year

# [2.1.1] - 2016-10-13
### Changed
- fallback to ajax if pjax not supported, prevents infinite refresh and still loads

# [2.1.0] - 2016-10-11
### Changed
- default load mask background color
### Removed
- support for no data template

# [2.0.0] - 2016-01-29
### Changed
- change the arguments accepted by plugins who implement ConfirmableMixin ( breaking change ). The wrapped ajax save
method shall always have the data to be saved as a first param.

# [1.8.0] - 2016-01-27
### Changed
- aborted xhr will now trigger a cancelled event instead of error, existing error handlers should continue to
behave as expected

# [1.7.4] - 2016-01-06
### Fixed
- EditableDropdownPlugin no longer updates the label if confirmation fails

# [1.7.3] - 2016-01-06
### Added
- EditableDropdownPlugin is now also Confirmable

# [1.7.2] - 2015-12-15
### Fixed
- lost plugin style which hides icon dropdown text labels, occurred after a move from styleguide

# [1.7.1] - 2015-11-30
### Removed
- white-space: nowrap removed as a default

# [1.7.0] - 2015-11-24
### Changed
- table layout is now fixed by default

### Removed
- app specific button and dropdown styles
- app specific link styles

# [1.6.4] - 2015-11-23
### Changed
  - Slight adjustments for table header borders, including a fix for 2px borders from bootstrap

# [1.6.3] - 2015-11-04
### Added
  - Formatter method to ajax mixin

# [1.6.2] - 2015-11-04
### Changed
  - remove row plugin now registers a click handler in order to work well with confirmable

# [1.6.1] - 2015-11-04
### Fixed
  - conflicting init method in confirmable mixin
  - mixin structure was incorrect

# [1.6.0] - 2015-11-04
### Added
  - confirmable mixin, works in conjunction with any plugin implementing a _save method

### Changed
  - remove row plugin now uses the confirmable mixin, a beforeSave function can be passed into the plugin config

# [1.5.2] - 2015-09-11
### Removed
  - debug flag from browserify build, fixes missing semi-colon in non-minified version

# [1.5.1] - 2015-09-09
### Added
  - bundle external search plugin as part of default build

# [1.5.0] - 2015-09-09
### Added
  - cell plugin mixin, sets default element, record, and option references
  - ajax cell mixin, provides an ajax post default with some events
  - editable dropdown, a plugin which posts information on change of a dropdown in a table cell
  - remove row, a plugin which posts an id/primary key to remove a row as a record from the data set
  - editable cell, a wrapper for x-editable to be used inside table cells

### Changed
  - project now uses browserify to bundle itself
  - separated widget construction into a utility module
  - search box widget code also separated, broken into new directory as an "external plugin" ( plugin that exists outside of the table )
  - table constructor is now exposed on window in addition to being a jquery plugin
  - new cell mixins/plugins are exposed through a PjaxTableShared object on window

# [1.4.10] - 2015-09-08
### Fixed
- Table cell sizing in broken in firefox by unnecessary vertical align

# [1.4.9] - 2015-08-21
### Added
- A load sync function. In addition to querySync ( to likely be renamed in the next major release ), the ability to
specify a function to be called on every load is desired. querySync currently executes once on page load. loadSync
applies to every load after that ( or initial page load if table data is lazy-loaded ).

# [1.4.8] - 2015-08-20
### Changed
- Reset the page to 1 when changing sort order

# [1.4.7] - 2015-08-14
### Fixed
- Fixes incorrect options reference leftover from revealing module

# [1.4.6] - 2015-08-10
### Changed
- Prevents retry on failed load

# [1.4.5] - 2015-08-06
### Changed
- Fixes incorrect reference of `$searchFilter` to `this._$searchFilter.val()` in the `PjaxTableSearch._onClickSearch()` method

# [1.4.4] - 2015-07-01
### Changed
- namespace updates to exports and test files

# [1.4.3] - 2015-06-30
### Added
- added an event object to plugins's `options`

# [1.4.2] - 2015-06-29
### Added
- added an optional function "syncQueryFn" for syncing the query state from attributes not explicit in the table js, e.g. search and filter status

# [1.4.1] - 2015-06-08
### Changed
- change data('url') to data('pjax-url') fixing a previous break in compatibility with python

# [1.4.0] - 2015-06-05
### Added
- auto init tables are now configurable through a global config object window.PjaxTableConfig

# [1.3.0] - 2015-06-04
### Added
- a public load method which resembles refresh but accepts load parameters to be used by ajax/pjax
- the public update method now accepts an additional param which is the load paramaters to be used by ajax/pjax

# [1.2.6] - 2015-06-04
### Added
- Added a queryState option so that you can initialize the table with a state that is separate from / in addition to the state synced from data attributes. The use case is for other
page components which drive table state but are not tied directly to the table, such as a date picker that provides a date range.

# [1.2.5] - 2015-05-28
### Changed
- changed the url fallback to be window.location.pathname ( the path without the query string ) which fixes a bug with query arg duplication. Bug was not completely addressed in 1.2.4

# [1.2.4] - 2015-05-28
### Changed
- changed the url fallback to be empty string ( the current base url ) which fixes a bug with query arg duplication

# [1.2.3] - 2015-05-11
### Changed

standalone/js/data.js
var data = [
	{"Heroes": "Griffith Shaffer", "Nemeses": "Anika M. Gilliam", "Nationality": "Malawi", "Cost to corrupt": "$38,426", "Email": "nec.cursus@magna.ca"},
	{"Heroes": "Germaine Kerr", "Nemeses": "Len B. Pacheco", "Nationality": "Mongolia", "Cost to corrupt": "$30,134", "Email": "rutrum.magna@massaSuspendisse.org"},
	{"Heroes": "Jameson Peterson", "Nemeses": "Talon P. Baxter", "Nationality": "Barbados", "Cost to corrupt": "$15,583", "Email": "Fusce.feugiat.Lorem@Crasdolordolor.edu"},
	{"Heroes": "Lane Leblanc", "Nemeses": "Lareina B. Bender", "Nationality": "Equatorial Guinea", "Cost to corrupt": "$88,569", "Email": "ipsum.porta.elit@Donecat.co.uk"},
	{"Heroes": "Tana Mcdowell", "Nemeses": "Hadassah B. Pierce", "Nationality": "Sri Lanka", "Cost to corrupt": "$53,880", "Email": "Vivamus.nisi.Mauris@erosProin.edu"},
	{"Heroes": "Yen Castro", "Nemeses": "Naomi X. Workman", "Nationality": "Korea, South", "Cost to corrupt": "$56,399", "Email": "laoreet.ipsum@Cumsociisnatoque.org"},
	{"Heroes": "Quinn Garza", "Nemeses": "Walker M. Strong", "Nationality": "Kiribati", "Cost to corrupt": "$55,735", "Email": "porttitor.vulputate@idenim.co.uk"},
	{"Heroes": "Ifeoma Santiago", "Nemeses": "Josephine G. Sweet", "Nationality": "Faroe Islands", "Cost to corrupt": "$42,306", "Email": "malesuada.fames.ac@sapien.ca"},
	{"Heroes": "Geoffrey Clements", "Nemeses": "Ulric S. Foreman", "Nationality": "American Samoa", "Cost to corrupt": "$96,893", "Email": "malesuada@magnisdis.net"},
	{"Heroes": "Tatyana Miranda", "Nemeses": "Latifah U. Weiss", "Nationality": "Serbia", "Cost to corrupt": "$94,087", "Email": "Lorem.ipsum@turpis.com"},
	{"Heroes": "Aquila Farrell", "Nemeses": "Ferris Y. Knowles", "Nationality": "Burkina Faso", "Cost to corrupt": "$42,249", "Email": "suscipit.est@Morbisitamet.com"},
	{"Heroes": "Heidi Wiggins", "Nemeses": "Wilma V. Golden", "Nationality": "Samoa", "Cost to corrupt": "$46,687", "Email": "ac.mattis.semper@penatibuset.org"},
	{"Heroes": "Rhiannon Watts", "Nemeses": "Akeem P. Hale", "Nationality": "Comoros", "Cost to corrupt": "$66,604", "Email": "sapien.Nunc@neceuismodin.net"},
	{"Heroes": "Jada Sweeney", "Nemeses": "Karly Q. Velasquez", "Nationality": "Turks and Caicos Islands", "Cost to corrupt": "$13,684", "Email": "dignissim.lacus@semperegestasurna.net"},
	{"Heroes": "Imelda Burnett", "Nemeses": "Burke Y. Roberson", "Nationality": "Cape Verde", "Cost to corrupt": "$73,285", "Email": "ligula.tortor.dictum@convallisantelectus.com"},
	{"Heroes": "Trevor Sawyer", "Nemeses": "Evangeline I. Pierce", "Nationality": "Northern Mariana Islands", "Cost to corrupt": "$78,856", "Email": "tellus@convallisin.net"},
	{"Heroes": "Leslie Meyer", "Nemeses": "Alden E. Fitzpatrick", "Nationality": "Lebanon", "Cost to corrupt": "$69,786", "Email": "nec@Morbivehicula.com"},
	{"Heroes": "Adrian Wall", "Nemeses": "Mariam V. Mcfadden", "Nationality": "Afghanistan", "Cost to corrupt": "$74,617", "Email": "vitae.risus.Duis@semper.org"},
	{"Heroes": "Rae Ball", "Nemeses": "Lamar E. Pena", "Nationality": "Ireland", "Cost to corrupt": "$31,406", "Email": "Vivamus.euismod@turpisegestas.org"},
	{"Heroes": "Ali Barrera", "Nemeses": "Amena P. Ayala", "Nationality": "Ethiopia", "Cost to corrupt": "$35,030", "Email": "urna.suscipit@et.edu"},
	{"Heroes": "Quinn Bruce", "Nemeses": "Casey K. Calderon", "Nationality": "Eritrea", "Cost to corrupt": "$72,400", "Email": "Integer.vulputate.risus@sagittisDuisgravida.net"},
	{"Heroes": "Chaim Sheppard", "Nemeses": "Palmer F. Goff", "Nationality": "Norway", "Cost to corrupt": "$44,160", "Email": "nunc.sed.libero@atiaculis.org"},
	{"Heroes": "Abbot Morrow", "Nemeses": "Lunea Z. Horn", "Nationality": "Guam", "Cost to corrupt": "$42,792", "Email": "quis@Inornaresagittis.net"},
	{"Heroes": "Sean Witt", "Nemeses": "Dillon C. Duran", "Nationality": "Grenada", "Cost to corrupt": "$7,306", "Email": "id.risus@malesuada.edu"},
	{"Heroes": "Latifah Clements", "Nemeses": "Kane B. Kirkland", "Nationality": "Suriname", "Cost to corrupt": "$12,583", "Email": "Sed@idante.ca"},
	{"Heroes": "Fulton Keller", "Nemeses": "Ella B. Estrada", "Nationality": "Bhutan", "Cost to corrupt": "$19,831", "Email": "torquent@pretiumaliquet.com"},
	{"Heroes": "Rae Stokes", "Nemeses": "Halla A. Becker", "Nationality": "Sierra Leone", "Cost to corrupt": "$49,155", "Email": "ut.mi@quisurnaNunc.edu"},
	{"Heroes": "Knox Sears", "Nemeses": "Pamela W. Whitney", "Nationality": "Monaco", "Cost to corrupt": "$93,569", "Email": "libero.Donec@ac.co.uk"},
	{"Heroes": "Sharon Mcmahon", "Nemeses": "Kellie F. Shannon", "Nationality": "Congo, the Democratic Republic of the", "Cost to corrupt": "$89,390", "Email": "tristique@utpharetra.net"},
	{"Heroes": "Amelia Graham", "Nemeses": "Uta C. Horn", "Nationality": "Germany", "Cost to corrupt": "$27,729", "Email": "Proin.dolor.Nulla@Phasellus.com"},
	{"Heroes": "Brent Delgado", "Nemeses": "Ella Y. Carroll", "Nationality": "Mozambique", "Cost to corrupt": "$71,166", "Email": "ut@Integeridmagna.com"},
	{"Heroes": "Jaquelyn Berger", "Nemeses": "Signe M. Dodson", "Nationality": "Kiribati", "Cost to corrupt": "$31,015", "Email": "et@luctus.net"},
	{"Heroes": "Caldwell Ellison", "Nemeses": "Owen Y. Kelly", "Nationality": "Kenya", "Cost to corrupt": "$83,955", "Email": "Aliquam.erat@elit.co.uk"},
	{"Heroes": "Miranda Castaneda", "Nemeses": "Frances W. Mcclure", "Nationality": "El Salvador", "Cost to corrupt": "$46,900", "Email": "adipiscing@Nuncmauris.ca"},
	{"Heroes": "Hedy Mcintyre", "Nemeses": "Kaitlin U. Baird", "Nationality": "American Samoa", "Cost to corrupt": "$70,016", "Email": "at.iaculis@acsem.co.uk"},
	{"Heroes": "Evan Alvarado", "Nemeses": "Ruth W. Chavez", "Nationality": "Sint Maarten", "Cost to corrupt": "$14,555", "Email": "orci@orci.org"},
	{"Heroes": "Louis Ingram", "Nemeses": "Lev K. Bean", "Nationality": "Suriname", "Cost to corrupt": "$73,215", "Email": "magna.malesuada@at.edu"},
	{"Heroes": "Jane Dorsey", "Nemeses": "Emmanuel E. King", "Nationality": "Niger", "Cost to corrupt": "$7,979", "Email": "Ut.tincidunt@Aliquameratvolutpat.edu"},
	{"Heroes": "Cain Stokes", "Nemeses": "Roary U. Colon", "Nationality": "Bosnia and Herzegovina", "Cost to corrupt": "$79,384", "Email": "neque.et@mipedenonummy.com"},
	{"Heroes": "Evan Mays", "Nemeses": "Molly S. Gray", "Nationality": "Jersey", "Cost to corrupt": "$43,390", "Email": "ultrices@ametornarelectus.com"},
	{"Heroes": "Herrod Vega", "Nemeses": "Imani H. Hampton", "Nationality": "Sudan", "Cost to corrupt": "$30,301", "Email": "elit@arcuvel.net"},
	{"Heroes": "Mari Barker", "Nemeses": "Bethany G. Mccall", "Nationality": "Latvia", "Cost to corrupt": "$84,554", "Email": "risus.Nulla.eget@orciquislectus.co.uk"},
	{"Heroes": "Phoebe Obrien", "Nemeses": "Dillon U. Gilbert", "Nationality": "Belgium", "Cost to corrupt": "$50,687", "Email": "Aenean.massa.Integer@lectusquismassa.co.uk"},
	{"Heroes": "Jescie Norris", "Nemeses": "Delilah L. Hopkins", "Nationality": "Ethiopia", "Cost to corrupt": "$95,180", "Email": "dolor.tempus.non@bibendumsed.org"},
	{"Heroes": "Regina Hall", "Nemeses": "Kelly V. Robles", "Nationality": "Hungary", "Cost to corrupt": "$42,018", "Email": "dis@Nullam.edu"},
	{"Heroes": "Solomon Hurley", "Nemeses": "Trevor D. Wade", "Nationality": "Lesotho", "Cost to corrupt": "$4,790", "Email": "dictum.Phasellus.in@tortorat.com"},
	{"Heroes": "Ori Hays", "Nemeses": "Melyssa B. Spencer", "Nationality": "Chad", "Cost to corrupt": "$4,283", "Email": "dictum.Proin@acturpis.edu"},
	{"Heroes": "Mara Bishop", "Nemeses": "Hamish U. Massey", "Nationality": "Costa Rica", "Cost to corrupt": "$8,433", "Email": "amet.lorem@nibhlaciniaorci.co.uk"},
	{"Heroes": "Alec Petty", "Nemeses": "Clio F. Thornton", "Nationality": "San Marino", "Cost to corrupt": "$84,974", "Email": "mi.Duis.risus@nec.ca"},
	{"Heroes": "Mannix Walsh", "Nemeses": "Ivor E. Suarez", "Nationality": "Gibraltar", "Cost to corrupt": "$34,312", "Email": "eget@consectetuereuismodest.net"},
	{"Heroes": "Judith Kramer", "Nemeses": "Sophia Z. Newman", "Nationality": "Cook Islands", "Cost to corrupt": "$22,470", "Email": "quis@sed.net"},
	{"Heroes": "Carolyn Quinn", "Nemeses": "Anjolie Y. Marsh", "Nationality": "Benin", "Cost to corrupt": "$9,908", "Email": "Etiam.gravida.molestie@egestasblandit.com"},
	{"Heroes": "Reed Wilkinson", "Nemeses": "Savannah P. Ellison", "Nationality": "Mozambique", "Cost to corrupt": "$11,345", "Email": "nonummy.ipsum.non@adipiscing.ca"},
	{"Heroes": "Yeo Murphy", "Nemeses": "Chaney A. Moreno", "Nationality": "Luxembourg", "Cost to corrupt": "$73,806", "Email": "sit.amet@diam.net"},
	{"Heroes": "Margaret Welch", "Nemeses": "Kimberley Y. Waller", "Nationality": "South Sudan", "Cost to corrupt": "$69,073", "Email": "magna@Pellentesque.com"},
	{"Heroes": "Myles Reilly", "Nemeses": "Beau N. Gray", "Nationality": "South Sudan", "Cost to corrupt": "$75,803", "Email": "lectus@Proinmi.net"},
	{"Heroes": "Denton Gill", "Nemeses": "Lavinia N. Deleon", "Nationality": "Ukraine", "Cost to corrupt": "$22,472", "Email": "vel@variusultricesmauris.ca"},
	{"Heroes": "Mikayla Nixon", "Nemeses": "Frances Z. Burgess", "Nationality": "Pakistan", "Cost to corrupt": "$70,794", "Email": "lacinia.at.iaculis@eratsemper.edu"},
	{"Heroes": "Adria Neal", "Nemeses": "Aimee N. Lawrence", "Nationality": "Central African Republic", "Cost to corrupt": "$19,408", "Email": "felis@etliberoProin.co.uk"},
	{"Heroes": "Zena Clarke", "Nemeses": "Charlotte B. Fletcher", "Nationality": "Moldova", "Cost to corrupt": "$83,237", "Email": "lorem.semper@Suspendissealiquet.ca"},
	{"Heroes": "Leigh Cole", "Nemeses": "Wang I. Garrison", "Nationality": "Lesotho", "Cost to corrupt": "$17,991", "Email": "Nullam.lobortis@quis.com"},
	{"Heroes": "Lilah Austin", "Nemeses": "Brielle B. Hyde", "Nationality": "Aruba", "Cost to corrupt": "$44,544", "Email": "Vestibulum.ante.ipsum@etmagnis.net"},
	{"Heroes": "Geraldine Rosa", "Nemeses": "Neil P. Copeland", "Nationality": "Mozambique", "Cost to corrupt": "$50,252", "Email": "feugiat@variuset.org"},
	{"Heroes": "Barclay Harris", "Nemeses": "Nomlanga R. English", "Nationality": "Bangladesh", "Cost to corrupt": "$29,652", "Email": "dignissim@nonummy.co.uk"},
	{"Heroes": "Roth Davenport", "Nemeses": "Thor T. Freeman", "Nationality": "Anguilla", "Cost to corrupt": "$81,200", "Email": "dis@seddictumeleifend.com"},
	{"Heroes": "Ryan Jackson", "Nemeses": "Ahmed F. Price", "Nationality": "New Zealand", "Cost to corrupt": "$60,556", "Email": "Duis.ac@bibendumfermentum.org"},
	{"Heroes": "Lucas Mccullough", "Nemeses": "Renee R. Hampton", "Nationality": "Romania", "Cost to corrupt": "$98,941", "Email": "mollis@auctor.edu"},
	{"Heroes": "Britanni Nichols", "Nemeses": "Owen H. Franks", "Nationality": "Bulgaria", "Cost to corrupt": "$21,456", "Email": "tellus.lorem.eu@faucibus.net"},
	{"Heroes": "Logan Anthony", "Nemeses": "Clinton P. Cochran", "Nationality": "Greece", "Cost to corrupt": "$89,432", "Email": "est@necurnaet.org"},
	{"Heroes": "Hayley Sawyer", "Nemeses": "Claire H. Sharpe", "Nationality": "Virgin Islands, United States", "Cost to corrupt": "$6,433", "Email": "quam.a@magna.net"},
	{"Heroes": "Warren Obrien", "Nemeses": "Cairo R. Hickman", "Nationality": "Romania", "Cost to corrupt": "$93,496", "Email": "Integer@hymenaeosMaurisut.edu"},
	{"Heroes": "Dana Barrera", "Nemeses": "Sigourney M. Mclaughlin", "Nationality": "Yemen", "Cost to corrupt": "$74,547", "Email": "ipsum.ac@ut.co.uk"},
	{"Heroes": "Allistair Buckner", "Nemeses": "Abraham D. Randolph", "Nationality": "Reunion", "Cost to corrupt": "$30,919", "Email": "lacus@Aliquam.edu"},
	{"Heroes": "Daryl Stein", "Nemeses": "Taylor Z. York", "Nationality": "Kenya", "Cost to corrupt": "$26,423", "Email": "tempor.diam.dictum@luctusaliquetodio.net"},
	{"Heroes": "Uriah Bell", "Nemeses": "Tamekah A. Mcgowan", "Nationality": "Kuwait", "Cost to corrupt": "$28,314", "Email": "vitae@vitae.net"},
	{"Heroes": "Hope Collins", "Nemeses": "Kristen U. Ramirez", "Nationality": "French Guiana", "Cost to corrupt": "$24,314", "Email": "eu.accumsan.sed@lectussit.com"},
	{"Heroes": "Ocean Dennis", "Nemeses": "Fleur B. Blankenship", "Nationality": "Ireland", "Cost to corrupt": "$65,825", "Email": "vehicula.et.rutrum@inhendrerit.edu"},
	{"Heroes": "Breanna Lyons", "Nemeses": "Evangeline K. Trevino", "Nationality": "Togo", "Cost to corrupt": "$52,936", "Email": "Lorem.ipsum@nibhenim.edu"},
	{"Heroes": "Dillon Wiggins", "Nemeses": "Xavier X. Henry", "Nationality": "Togo", "Cost to corrupt": "$30,969", "Email": "non.magna@rhoncusNullam.co.uk"},
	{"Heroes": "Melissa Harrison", "Nemeses": "Velma Q. Jordan", "Nationality": "Djibouti", "Cost to corrupt": "$10,613", "Email": "elementum.purus.accumsan@Curabitur.ca"},
	{"Heroes": "Cody Guthrie", "Nemeses": "Shelly C. Page", "Nationality": "Albania", "Cost to corrupt": "$98,637", "Email": "ut.eros.non@ligula.org"},
	{"Heroes": "Wylie Gilmore", "Nemeses": "Anika Y. Maldonado", "Nationality": "Costa Rica", "Cost to corrupt": "$0,862", "Email": "turpis.non.enim@Namconsequatdolor.net"},
	{"Heroes": "Caryn Barton", "Nemeses": "Patrick H. Ruiz", "Nationality": "Finland", "Cost to corrupt": "$98,507", "Email": "ornare.egestas@risus.net"},
	{"Heroes": "Idola Hickman", "Nemeses": "Suki H. Pope", "Nationality": "Solomon Islands", "Cost to corrupt": "$49,157", "Email": "mi@vitaealiquetnec.net"},
	{"Heroes": "Jolene Booker", "Nemeses": "Ginger V. Lawrence", "Nationality": "Wallis and Futuna", "Cost to corrupt": "$93,098", "Email": "augue.scelerisque.mollis@natoquepenatibuset.org"},
	{"Heroes": "Shellie Hoffman", "Nemeses": "Griffith Z. Anthony", "Nationality": "Benin", "Cost to corrupt": "$39,742", "Email": "eget.ipsum@gravidamaurisut.ca"},
	{"Heroes": "Sean Levy", "Nemeses": "Charity F. Randolph", "Nationality": "Saint Lucia", "Cost to corrupt": "$19,379", "Email": "nisi.a.odio@variusNam.co.uk"},
	{"Heroes": "Laurel Powell", "Nemeses": "Debra E. Richardson", "Nationality": "Norway", "Cost to corrupt": "$5,491", "Email": "magna.Duis@cubilia.co.uk"},
	{"Heroes": "Felicia Newman", "Nemeses": "Stuart R. Dalton", "Nationality": "Korea, South", "Cost to corrupt": "$95,576", "Email": "consectetuer.adipiscing.elit@Sed.edu"},
	{"Heroes": "Natalie Hensley", "Nemeses": "Brianna M. Douglas", "Nationality": "Central African Republic", "Cost to corrupt": "$78,576", "Email": "Phasellus.elit.pede@Sedeu.net"},
	{"Heroes": "Samson Rutledge", "Nemeses": "Griffith X. Reid", "Nationality": "Saint Lucia", "Cost to corrupt": "$65,095", "Email": "placerat.orci@pharetraNam.edu"},
	{"Heroes": "Blair Cooley", "Nemeses": "Price A. Blankenship", "Nationality": "Kuwait", "Cost to corrupt": "$17,176", "Email": "eu.tellus.Phasellus@risusat.ca"},
	{"Heroes": "Petra Church", "Nemeses": "Leilani B. Pena", "Nationality": "Cape Verde", "Cost to corrupt": "$70,805", "Email": "lorem.auctor.quis@Nunclaoreetlectus.edu"},
	{"Heroes": "Amir Kirby", "Nemeses": "Ebony D. Evans", "Nationality": "Zambia", "Cost to corrupt": "$53,197", "Email": "nulla@erategetipsum.com"},
	{"Heroes": "Raya Jenkins", "Nemeses": "Sawyer W. Bishop", "Nationality": "Taiwan", "Cost to corrupt": "$73,628", "Email": "mauris.Integer.sem@nonmagnaNam.net"},
	{"Heroes": "Kyla Owens", "Nemeses": "Yasir I. Pena", "Nationality": "Malawi", "Cost to corrupt": "$1,597", "Email": "malesuada@eu.edu"},
	{"Heroes": "Carol Mooney", "Nemeses": "Vernon E. Ramos", "Nationality": "Nigeria", "Cost to corrupt": "$97,507", "Email": "ante.Maecenas.mi@eget.edu"},
	{"Heroes": "Damon Casey", "Nemeses": "Maggie W. Nelson", "Nationality": "Guinea-Bissau", "Cost to corrupt": "$92,156", "Email": "rutrum.eu.ultrices@Cumsociisnatoque.net"},
	{"Heroes": "Nell Gonzales", "Nemeses": "Warren D. Hobbs", "Nationality": "Chad", "Cost to corrupt": "$57,262", "Email": "nonummy@ut.net"},
	{"Heroes": "Clare Craig", "Nemeses": "Edward X. Rogers", "Nationality": "Grenada", "Cost to corrupt": "$22,461", "Email": "blandit.at.nisi@nuncQuisque.com"}
];
